### PR TITLE
bugfix: infinite diff. for burn alert and trigger pagerduty recipients if severity not specified

### PR DESCRIPTION
**Pull request by jharley**
Discovered while investigating #200

### CODE REVIEW

File: honeycombio/resource_burn_alert_test.go [hunk 1]
```diff
@@ -135,10 +135,7 @@ resource "honeycombio_burn_alert" "test" {
 
   recipient {
     id = honeycombio_pagerduty_recipient.test.id
-
-    notification_details {
-      pagerduty_severity = "critical"
-    }
+    // default value is critical
```
Comment (dstrelau) [inline honeycombio/resource_burn_alert_test.go:138]: now that you removed the details block, "value" is unclear. Maybe just expand this comment a bit?

File: honeycombio/type_helpers.go [hunk 2]
```diff
@@ -156,10 +156,10 @@ func tpmToFloat(t int) float64 {
 }
 
 func flattenNotificationRecipients(rs []honeycombio.NotificationRecipient) []map[string]interface{} {
-	result := make([]map[string]interface{}, len(rs))
+	result := []map[string]interface{}{}
```
Comment (dstrelau) [inline honeycombio/type_helpers.go:159]: Why did you change this slice construction? The `make` that was here is more idiomatic, IMO.

File: honeycombio/type_helpers.go [hunk 3]
```diff
@@ -193,11 +194,19 @@ func expandNotificationRecipients(s []interface{}) []honeycombio.NotificationRec
...
-				recipients[i].Details = &honeycombio.NotificationRecipientDetails{
+				rcpt.Details = &honeycombio.NotificationRecipientDetails{
 					PDSeverity: honeycombio.PagerDutySeverity(s.(string)),
 				}
 			}
+		} else {
+			// set default notification details
+			if rcpt.Type == honeycombio.RecipientTypePagerDuty {
+				rcpt.Details = &honeycombio.NotificationRecipientDetails{
+					PDSeverity: honeycombio.PDSeverityCRITICAL,
```
Comment (dstrelau) [inline honeycombio/type_helpers.go:205]: Where is this used such that we need to hard-code the default here. It'd be nice if we could rely on the API to provide this and not have to duplicate that logic.

File: honeycombio/type_helpers.go (hunk 2 above)
Comment (jharley) [reply honeycombio/type_helpers.go:159]: Ah, fair. I overcorrected for inserting by index (instead of append). I can flip this back 👍🏻

File: honeycombio/type_helpers.go (hunk 3 above)
Comment (jharley) [reply honeycombio/type_helpers.go:205]: Nice catch! This is basically unused because we're already reading back from the API when the function is called

**Review by dstrelau — APPROVED** [review]
